src/udx.c

```diff
@@ -1195,52 +1173,8 @@ on_uv_poll (uv_poll_t *handle, int status, int events) {
     }
   }
 
-  while (events & UV_WRITABLE && socket->send_queue.len > 0) {
-    udx_packet_t *pkt = (udx_packet_t *) udx__fifo_shift(&(socket->send_queue));
-    if (pkt == NULL) continue;
-
-    bool adjust_ttl = pkt->ttl > 0 && socket->ttl != pkt->ttl;
-
-    if (adjust_ttl) uv_udp_set_ttl((uv_udp_t *) socket, pkt->ttl);
-
-    if (socket->family == 6 && pkt->dest.ss_family == AF_INET) {
-      addr_to_v6((struct sockaddr_in *) &(pkt->dest));
-      pkt->dest_len = sizeof(struct sockaddr_in6);
-    }
-
-    size = udx__sendmsg(socket, pkt->bufs, pkt->bufs_len, (struct sockaddr *) &(pkt->dest), pkt->dest_len);
-
-    if (adjust_ttl) uv_udp_set_ttl((uv_udp_t *) socket, socket->ttl);
-
-    if (size == UV_EAGAIN) {
-      udx__fifo_undo(&(socket->send_queue));
-      break;
-    }
-
-    assert(pkt->status == UDX_PACKET_SENDING);
-    pkt->status = UDX_PACKET_INFLIGHT;
-    pkt->transmits++;
-    pkt->time_sent = uv_hrtime() / 1e6;
-
-    int type = pkt->type;
-
-    if (type & UDX_PACKET_CALLBACK) {
-      trigger_send_callback(socket, pkt);
-      // TODO: watch for re-entry here!
-    }
-
-    if (type & UDX_PACKET_FREE_ON_SEND) {
-      free(pkt);
-    }
-
-    // queue another write, might be able to do this smarter...
-    if (socket->send_queue.len > 0) continue;
-
-    // if the socket is under closure, we need to trigger shutdown now since no important writes are pending
-    if (socket->status & UDX_SOCKET_CLOSING) {
-      close_handles(socket);
-      return;
-    }
+  if (events & UV_WRITABLE) {
+    udx__on_write_ready(socket);
```

Comment:
i was a bit confused by the name. wdyt about something like `udx__on_writable` (super nit department!)